aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,639 @@
+
+    /// Can only propose a start time in future
+    const EPROPOSAL_START_TIME_SHOULD_IN_FUTURE: u64 = 7;
+
+    /// Cannot only use membership token of the DAO
+    const ENOT_A_MEMBER_TOKEN: u64 = 8;
+
+    /// String length exceeds limits
+    const ESTRING_TOO_LONG: u64 = 9;
+
+    /// Proposal ended
+    const EPROPOSAL_ENDED: u64 = 10;
+
+    /// Propsoal already resolved
+    const EPROPOSAL_RESOLVED: u64 = 11;
+
+    /// Denominator should be bigger than 0
+    const EDEMONINATOR_CANNOT_BE_ZERO: u64 = 12;
+
+    /// Denominator should be bigger or equal to numerator
+    const ENUMERATOR_BIGGER_THAN_DENOMINATOR: u64 = 13;
+
+    /// NOT ENOUGH PEOPLE VOTED
+    const ENOT_ENOUGH_TOKEN_VOTED: u64 = 14;
+
+    /// TOKEN VOTED FOR PROPOSAL
+    const ETOKEN_ALREADY_VOTED: u64 = 15;
+
+    /// Resource Account Doesn't exist
+    const ERESOURCE_ACCT_NOT_EXIST: u64 = 16;
+
+    /// Proposal resource not created
+    const EPRPOSALS_NOT_EXIST_AT_ADDRESS: u64 = 17;
+
+    /// Proposal id doesn't exist
+    const EPRPOSAL_ID_NOT_EXIST: u64 = 18;
+
+
+    /// Constants
+    const PROPOSAL_UNRESOLVED: u8 = 0;
+    const PROPOSAL_RESOLVED_PASSED: u8 = 1;
+    const PROPOSAL_RESOLVED_NOT_PASSED: u8 = 2;
+
+    const DEFALT_VOTING_WEIGHT: u64 = 1;
+
+    struct DAO has key {
+        /// Name of the DAO
+        name: String,
+        /// The threshold that the proposal can resolve. For example, 50% means at least 50% of NFT holder should vote to resolve the proposal
+        resolve_threshold: ResolveThreshold,
+        /// The NFT Collection that is used to govern the DAO
+        governance_token: GovernanceToken, // This is the governance token or NFT used for voting
+        /// The address of the creator account this DAO owns. THis account is a resource account that this DAO is an adminstrator of the controlled account
+        controlled_account: address, // This is the nft creator account that this DAO governs, this accout should offer it signer cap to DAO account
+        /// The voting duration in secs
+        voting_duration: u64,
+        /// Total number of eligible member NFTs
+        total_supply: u64,
+        /// Minimum weight for proposal
+        min_proposal_weight: u64,
+        /// Proposal counter
+        cur_largest_proposal_id: u64,
+    }
+
+    struct ResolveThreshold has store {
+        numerator: u64,
+        denominator: u64,
+    }
+
+    /// The collection should only contains NFTs, where all token name only has 1 maximal and immutable
+    /// The total supply is fixed with the token names.
+    struct GovernanceToken has store {
+        /// The creator address of this NFT collection
+        creator: address,
+        /// The collection name
+        collection: String,
+        /// Eligible token_ids that can vote. Value of the table is the weights of the token
+        member_tokens: Table<TokenId, u64>
+    }
+
+    /// All proposals
+    struct Proposals has key {
```

Comment:
Does this need to be a separate struct from the DAO struct?

Reply:
I don't see the data access of DAO and proposal calls are same. For example, If I just want to read a proposal, I don't want to borrow the entire DAO struct and local the proposal from it.  verse versa, if I read DAO configs, I don't want to load an entire blob of resources containing all the proposals.

Reply:
Alright that makes sense 👍